### PR TITLE
feat: docs addition for RS SearchBox

### PR DESCRIPTION
**Pull request by mohdashraf010897**
**PR Type** For ReactiveSearch, we created a Searchbox component. This PR adds docs support for the same.

- [x]  Searchbox support in React
- [ ]  Searchbox support in Vue

Implementation  PR: https://github.com/appbaseio/reactivesearch/pull/1806

[Notion Link](https://www.notion.so/appbase/8648760a66da4dcfa5ebcefd2e439ddc)

### CODE REVIEW

File: content/docs/reactivesearch/v3/search/searchbox.md [hunk 1]
```diff
@@ -0,0 +1,837 @@
...
+
+```js
+<SearchBox componentId="SearchBoxSensor" dataField={['group_venue', 'group_city']} />
+```
+
+### Usage With All Props
+
+```js
+<SearchBox
+	componentId="SearchBoxSensor"
```
Comment (siddharthlatest) [inline content/docs/reactivesearch/v3/search/searchbox.md]: @mohdashraf010897 Check the indent here.

File: src/data/sidebars/web-reactivesearch.yaml [hunk 2]
```diff
@@ -63,6 +63,8 @@
               link: /docs/reactivesearch/v3/search/datasearch/
             - title: CategorySearch
               link: /docs/reactivesearch/v3/search/categorysearch/
+            - title: SearchBox
```
Comment (siddharthlatest) [inline src/data/sidebars/web-reactivesearch.yaml]: @mohdashraf010897 Move this above DataSearch.

File: content/docs/reactivesearch/v3/search/searchbox.md [hunk 3]
```diff
@@ -0,0 +1,837 @@
...
+
+-   **categoryField** `string` [optional]
+    Data field whose values are used to provide category specific suggestions.
+
+## Demo
+
+<br />
+
+<!-- Example will be added once the implementation PR is merged -->
+<iframe src="https://codesandbox.io/embed/github/appbaseio/reactivesearch/tree/next/packages/web/examples/DataSearch" style="width:100%; height:500px; border:0; border-radius: 4px; overflow:hidden;" sandbox="allow-modals allow-forms allow-popups allow-scripts allow-same-origin"></iframe>
```
Comment (siddharthlatest) [inline content/docs/reactivesearch/v3/search/searchbox.md]: @mohdashraf010897 We should update this example before merging this PR.

**Review by siddharthlatest — CHANGES_REQUESTED** [review]

File: content/docs/reactivesearch/v3/search/searchbox.md [hunk 4]
```diff
@@ -0,0 +1,842 @@
...
+```
+
+-   **enablePredictiveSuggestions** `Boolean` [optional]
+    Defaults to `false`. When set to `true`, it predicts the next relevant words from a field's value based on the search query typed by the user. When set to false (default), the matching document field's value would be displayed.
+
+-   **downShiftProps** `Object` [optional]
+    allow passing props directly to the underlying `Downshift` component. You can read more about Downshift props [here](https://github.com/paypal/downshift#--downshift-------).
+-   **fieldWeights** `Array` [optional] <mark color="yellow">[deprecated]</mark>
+    set the search weight for the database fields, useful when dataField is an Array of more than one field. This prop accepts an array of numbers. A higher number implies a higher relevance weight for the corresponding field in the search results.
+> Note: The `fieldWeights` property has been marked as deprecated in <b>v3.21.0</b> of ReactiveSearch and <b>v7.47.0</b> of Appbase and would be removed in the next major release. We recommend you to use the [dataField](/docs/search/reactivesearch-api/reference/#datafield) property to define the weights.
```
Comment (bietkul) [inline content/docs/reactivesearch/v3/search/searchbox.md:220]:
```suggestion
> Note: The `fieldWeights` property has been marked as deprecated in <b>v3.21.0</b> of ReactiveSearch and <b>v7.47.0</b> of Appbase and would be removed in the next major release. We recommend you to use the [dataField](/docs/search/reactivesearch-api/reference/#datafield) property to define the weights.

```

**Review by bietkul — CHANGES_REQUESTED** [review]

File: content/docs/reactivesearch/v3/search/searchbox.md [hunk 5]
```diff
@@ -0,0 +1,843 @@
...
+    set the title of the component to be shown in the UI.
+-   **defaultValue** `string` [optional]
+    set the initial search query text on mount.
+-   **value** `string` [optional]
+    sets the current value of the component. It sets the search query text (on mount and on update). Use this prop in conjunction with the `onChange` prop.
+-   **enableSynonyms** `bool` [optional]
+    Defaults to `true`, can be used to `disable/enable` the synonyms behavior for the search query. Read more about it [here](/docs/search/reactivesearch-api/reference/#enablesynonyms)
+    > Note:
+    >
+    > This property only works with [ReactiveSearch API](/docs/search/reactivesearch-api/) i.e when `enableAppbase` is set to `true` in `ReactiveBase` component.
```
Comment (bietkul) [inline content/docs/reactivesearch/v3/search/searchbox.md:156]:
```suggestion
    > Note:
    >
    > This property only works with [ReactiveSearch API](/docs/search/reactivesearch-api/) i.e when `enableAppbase` is set to `true` in `ReactiveBase` component.
   
```

**Review by bietkul — CHANGES_REQUESTED** [review]

**Pull request merged**